### ISSUE TITLE
OpenJ9 Java11 failure when starting Jenkins app

### ISSUE THREAD

**Issue by gjdeval** [eclipse/openj9#6884]
Java -version output
--------------------

openjdk version "11.0.4" 2019-07-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.4+11)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.15.1, JRE 11 Linux amd64-64-Bit Compressed References 20190717_286 (JIT enabled, AOT enabled)
OpenJ9   - 0f66c6431
OMR      - ec782f26
JCL      - fa49279450 based on jdk-11.0.4+11)

Summary of problem
------------------

Jenkins app (in Liberty server) fails to initialize properly with OpenJ9 11 - execution terminates with a traceassert. 

Here's an example of the error output in the console.log (stderr):

WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.thoughtworks.xstream.converters.reflection.AbstractAttributedCharacterIteratorAttributeConverter (file:/opt/IBM/Liberty/CL-19009-190823-1310/wlp/usr/servers/jenkins/workarea/org.eclipse.osgi/64/data/cache/com.ibm.ws.app.manager_4/.cache/WEB-INF/lib/xstream-1.4.7-jenkins-1.jar) to method java.text.AttributedCharacterIterator$Attribute.getName()
WARNING: Please consider reporting this to the maintainers of com.thoughtworks.xstream.converters.reflection.AbstractAttributedCharacterIteratorAttributeConverter
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
22:45:19.323 0x187bd00 omrport.359    *   ** ASSERTION FAILED ** at ../../omr/port/common/omrmemtag.c:145: ((memoryCorruptionDetected))

Diagnostic files
----------------

Will upload cores etc. shortly

**Comment by gjdeval** [comment]
Javacore, Snap, and core files (jextracted) uploaded to box
https://ibm.ent.box.com/folder/85928679740

**Comment by gjdeval** [comment]
FYI, the same type of failure is seen with the latest OpenJ9-12 release build

openjdk version "12.0.2" 2019-07-16
OpenJDK Runtime Environment AdoptOpenJDK (build 12.0.2+10)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.15.1, JRE 12 Linux amd64-64-Bit
OpenJ9   - 0f66c6431
OMR      - ec782f26
JCL      - 06c2cc3322 based on jdk-12.0.2+10)

**Comment by gjdeval** [comment]
FYI the traceassert error does not appear with the latest OpenHS-11 build

openjdk version "11.0.4" 2019-07-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.4+11)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.4+11, mixed mode)

A similar warning appears in the console.log
~~~~~~~~~~~~~
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.thoughtworks.xstream.converters.reflection.AbstractAttributedCharacterIteratorAttributeConverter (file:/opt/IBM/Liberty/OL-19009-190823-1310/wlp/usr/servers/jenkins/workarea/org.eclipse.osgi/64/data/cache/com.ibm.ws.app.manager_4/.cache/WEB-INF/lib/xstream-1.4.7-jenkins-1.jar) to method java.text.AttributedCharacterIterator$Attribute.getName()
WARNING: Please consider reporting this to the maintainers of com.thoughtworks.xstream.converters.reflection.AbstractAttributedCharacterIteratorAttributeConverter
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Jenkins home directory: /root/.jenkins found at: $user.home/.jenkins
~~~~~~~~~~~~~
but it is not followed by the memory corruption assert error.

**Comment by DanHeidinga** [comment]
Debug log:
```
$ docker run --rm -it -v `pwd`:/files adoptopenjdk/openjdk12-openj9 jdmpview -core /files/core.20190828.184452.8658.0001.dmp.zip 

> !findvm
!j9javavm 0x00007FF264043230
> !j9javavm 0x00007FF264043230 | grep port
	0x20: struct J9PortLibrary* portLibrary = !j9portlibrary 0x00007FF26AEA1300
	0x4a78: struct _jmethodID* addExports = !_jmethodid 0x0000000000000000
	0x4ae0: struct J9PortVmemIdentifier exclusiveGuardPage = !j9portvmemidentifier 0x00007FF264047D10

> !j9portlibrary 0x00007FF26AEA1300 | grep omr
	0x0: struct OMRPortLibrary omrPortLibrary = !omrportlibrary 0x00007FF26AEA1300

> !omrportlibrary 0x00007FF26AEA1300 | grep Globals
	0x0: struct OMRPortLibraryGlobalData* portGlobals = !omrportlibraryglobaldata 0x00007FF264008E40

> !omrportlibraryglobaldata 0x00007FF264008E40
OMRPortLibraryGlobalData at 0x7ff264008e40 {
  Fields for OMRPortLibraryGlobalData:
	0x0: void* corruptedMemoryBlock = !j9x 0x00007FF10C03E470  /////<----  Here's the bad block
	0x8: struct J9PortControlData control = !j9portcontroldata 0x00007FF264008E48
	0x30: struct J9NLSDataCache nls_data = !j9nlsdatacache 0x00007FF264008E70
	0x8b0: U64 tls_key = 0x0000000000000001 (1)
	0x8b8: pthread_mutex_t tls_mutex = !pthread_mutex_t 0x00007FF2640096F8
	0x8e0: void* buffer_list = !j9x 0x00007FF0440AB9B0
	0x8e8: void* procSelfMap = !j9x 0x00007FF264009A80
	0x8f0: struct OMRPortPlatformGlobals platformGlobals = !omrportplatformglobals 0x00007FF264009730
	0xb78: struct OMRMemCategory unknownMemoryCategory = !omrmemcategory 0x00007FF2640099B8
	0xba8: struct OMRMemCategory portLibraryMemoryCategory = !omrmemcategory 0x00007FF2640099E8
	0xbd8: U64 disableEnsureCap32 = 0x0000000000000000 (0)
	0xbe0: struct OMRMemCategory unusedAllocate32HeapRegionsMemoryCategory = !omrmemcategory 0x00007FF264009A20
	0xc10: U64 vmemAdviseOSonFree = 0x0000000000000001 (1)
	0xc18: U64 vectorRegsSupportOn = 0x0000000000000000 (0)
	0xc20: U64 userSpecifiedCPUs = 0x0000000000000000 (0)
	0xc28: U64 vmemEnableMadvise = 0x0000000000000000 (0)

**Comment by hangshao0** [comment]
Yes, it looks like #6549 to me. Allocating at the same place (shared.c:244) with size zero.

J9MemTag at 0x7ff10c03e450 {
  Fields for J9MemTag:
	0x0: U32 eyeCatcher = 0xB1234567 (2971878759)
	0x4: U32 sumCheck = 0x8ADA6F25 (2329571109)
	**0x8: U64 allocSize = 0x0000000000000000 (0)**
	0x10: const U8* callSite = !j9x 0x00007FF25D134B83
	0x18: struct OMRMemCategory* category = !omrmemcategory 0x00007FF26AE9FA60
}

**Comment by gjdeval** [comment]
Confirmed that this error no longer appears in the latest nightly openJ9-11 build
```openjdk version "11.0.5" 2019-10-15
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.5+4-201909021901)
Eclipse OpenJ9 VM AdoptOpenJDK (build master-e2e21457f, JRE 11 Linux amd64-64-Bit Compressed References 20190902_324 (JIT enabled, AOT enabled)
OpenJ9   - e2e21457f
OMR      - 5628a974
JCL      - 259a016054 based on jdk-11.0.5+4)```

**Comment by pshipton** [comment]
The next release for Java 11 will contain the fix, which is the 0.17 release.
